File: backend/app/schemas/resume.py
"""Resume-related Pydantic schemas."""

from datetime import datetime
from typing import Any, Dict, List, Optional
from uuid import UUID

from pydantic import BaseModel, Field


class CandidateCreate(BaseModel):
    name: str = Field(..., max_length=100)
    email: Optional[str] = Field(None, max_length=255)
    phone: Optional[str] = Field(None, max_length=30)
    current_role: Optional[str] = Field(None, max_length=200)
    years_of_exp: Optional[int] = Field(None, ge=0)
    notes: Optional[str] = None


class CandidateResponse(BaseModel):
    id: UUID
    name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    current_role: Optional[str] = None
    years_of_exp: Optional[int] = None
    notes: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class ResumeUploadResponse(BaseModel):
    id: UUID
    file_name: str
    file_size_bytes: int
    mime_type: str
    ocr_status: str
    ocr_error_msg: Optional[str] = None
    ocr_raw_text: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class ResumeResponse(BaseModel):
    id: UUID
    candidate_id: Optional[UUID] = None
    file_name: str
    file_size_bytes: int
    mime_type: str
    ocr_status: str
    ocr_raw_text: Optional[str] = None
    ocr_error_msg: Optional[str] = None
    parsed_data: Optional[Dict[str, Any]] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ResumeAnalysisResponse(BaseModel):
    id: UUID
    resume_id: UUID
    agent_model: str
    analysis_json: Dict[str, Any]
    tokens_used: Optional[int] = None
    processing_ms: Optional[int] = None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/resume.ts
export interface Candidate {
  id: string
  name: string
  email?: string
  phone?: string
  currentRole?: string
  yearsOfExp?: number
  notes?: string
  createdAt: string
}

export interface ResumeUploadResponse {
  id: string
  fileName: string
  fileSizeBytes: number
  mimeType: string
  ocrStatus: string
  createdAt: string
}

export interface ResumeInfo {
  id: string
  candidateId?: string
  fileName: string
  fileSizeBytes: number
  mimeType: string
  ocrStatus: string
  ocrRawText?: string
  ocrErrorMsg?: string
  parsedData?: ParsedResumeData
  createdAt: string
  updatedAt: string
}

export interface ParsedResumeData {
  name?: string
  email?: string
  phone?: string
  education?: Education[]
  experience?: WorkExperience[]
  skills?: string[]
  projects?: Project[]
}

export interface Education {
  school: string
  degree: string
  year: number
}

export interface WorkExperience {
  company: string
  role: string
  duration: string
  description: string
}

export interface Project {
  name: string
  description: string
  techStack: string[]
}

export interface ResumeAnalysis {
  id: string
  resumeId: string
  agentModel: string
  analysisJson: ResumeAnalysisData
  tokensUsed?: number
  processingMs?: number
  createdAt: string
}

export interface ResumeAnalysisData {
  overallAssessment: string
  strengths: string[]
  weaknesses: string[]
  skillMatch: Record<string, number>
  experienceRelevanceScore: number
  projectHighlights: Array<{ project: string; depthAnalysis: string }>
  suggestedQuestions: string[]
  redFlags: string[]
  tokensUsed: number
  processingMs: number
}
